src/App.tsx

```diff
@@ -4,32 +4,33 @@ import logo from "../public/FischLogo.svg";
 
 import styles from "./App.module.css";
 import { Card } from "./components/Card";
-import { useRef, useState } from "react";
+import { JSX, useState } from "react";
+import { NewCard } from "./components/NewCard";
 
-function App() {
-  const [cardsAmount, setCardsAmount] = useState(1)
-  const [cards, setCards] = useState([{}]);
+interface AppProps {
+  basicCards?: { face: string; back: string }[];
+}
+function App(props: AppProps) {
+  let basicCards: JSX.Element[] = [];
 
-  const handleAddCardButtonClick = () => {
-    setCardsAmount(cardsAmount+1)
-    setCards(prevCards => [...prevCards, {}]);
-  };
-  
+  props.basicCards && props.basicCards.forEach((cardData, cardIndex) => {
+    basicCards.push(<Card faceValue={cardData.face} flipValue={cardData.back} key={cardIndex+1} />)
+  });
 
-  const renderCards = () => {
-    return cards.map(() => (
-      <Card cardsAmount={cardsAmount} setCardsAmount={setCardsAmount} />
-    ));
+  const [cardsArray, setCardsArray] = useState(basicCards)
+
+  const handleAddCardButtonClick = () => {
+    setCardsArray([<NewCard key={cardsArray.length+1} />, ...cardsArray])
```

Comment:
**issue:** You should not set array of elements, but an array of objects. This is not correct React approach. Check it out here: https://react.dev/learn/rendering-lists